src/function/vector_list_functions.cpp

```diff
@@ -94,6 +95,32 @@ vector_function_definitions ListCreationVectorFunction::getDefinitions() {
     return result;
 }
 
+std::unique_ptr<FunctionBindData> ListRangeVectorFunction::bindFunc(
+    const binder::expression_vector& arguments, kuzu::function::FunctionDefinition* definition) {
+    auto varListTypeInfo =
+        std::make_unique<VarListTypeInfo>(std::make_unique<LogicalType>(LogicalTypeID::INT64));
+    auto resultType = LogicalType{LogicalTypeID::VAR_LIST, std::move(varListTypeInfo)};
+    return std::make_unique<FunctionBindData>(resultType);
+}
+
+vector_function_definitions ListRangeVectorFunction::getDefinitions() {
+    vector_function_definitions result;
+    // start, end
+    result.push_back(std::make_unique<VectorFunctionDefinition>(LIST_RANGE_FUNC_NAME,
```

Comment:
Take a look at how do we register `ADD`. I think you want to register 
```
range(int64, int64)
range(int32, int32)
range(int16, int16)
range(int8, int8)
... potentially unsigned int
```

This should save a reasonable amount of casting. If you only register 64, 64 then we will be casting for all non-64, non-64 input. The reason we don't register everything if because that will explode operation implementation.